crates/pop-parachains/src/build.rs

```diff
@@ -1,13 +1,498 @@
+	let raw_chain_spec = path.unwrap_or(Path::new("./")).join(chain_spec_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"build-spec",
+			"--chain",
+			&plain_chain_spec.display().to_string(),
+			"--disable-default-bootnode",
+			"--raw",
+		],
+	)
+	.stdout_path(raw_chain_spec.clone())
+	.run()?;
+	Ok(raw_chain_spec)
+}
+
+/// Export the WebAssembly runtime for the parachain.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `chain_spec` - Location of the raw chain specification file.
+/// * `binary_path` - The path to the node binary executable that contains the `export-genesis-wasm` command.
+/// * `wasm_file_name` - The name of the wasm runtime file to be generated.
+pub fn export_wasm_file(
+	path: Option<&Path>,
+	chain_spec: &Path,
+	binary_path: &Path,
+	wasm_file_name: &str,
+) -> Result<PathBuf, Error> {
+	if !chain_spec.exists() {
+		return Err(Error::MissingChainSpec(chain_spec.display().to_string()));
+	}
+	check_command_exists(&binary_path, "export-genesis-wasm")?;
+	let wasm_file = path.unwrap_or(Path::new("./")).join(wasm_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"export-genesis-wasm",
+			"--chain",
+			&chain_spec.display().to_string(),
+			&wasm_file.display().to_string(),
+		],
+	)
+	.run()?;
+	Ok(wasm_file)
+}
 
+/// Generate the parachain genesis state.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `chain_spec` - Location of the raw chain specification file.
+/// * `binary_path` - The path to the node binary executable that contains the `export-genesis-state` command.
+/// * `genesis_file_name` - The name of the genesis state file to be generated.
+pub fn generate_genesis_state_file(
+	path: Option<&Path>,
+	chain_spec: &Path,
+	binary_path: &Path,
+	genesis_file_name: &str,
+) -> Result<PathBuf, Error> {
+	if !chain_spec.exists() {
+		return Err(Error::MissingChainSpec(chain_spec.display().to_string()));
+	}
+	check_command_exists(&binary_path, "export-genesis-state")?;
+	let genesis_file = path.unwrap_or(Path::new("./")).join(genesis_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"export-genesis-state",
+			"--chain",
+			&chain_spec.display().to_string(),
+			&genesis_file.display().to_string(),
+		],
+	)
+	.run()?;
+	Ok(genesis_file)
+}
+
+/// Parses the node name from the Cargo.toml file located in the project path.
+fn parse_node_name(node_path: &Path) -> Result<String, Error> {
+	let cargo_toml = node_path.join("Cargo.toml");
+	let contents = std::fs::read_to_string(&cargo_toml)?;
+	let config = contents.parse::<DocumentMut>().map_err(|err| Error::TomlError(err.into()))?;
+	let name = config
+		.get("package")
+		.and_then(|i| i.as_table())
+		.and_then(|t| t.get("name"))
+		.and_then(|i| i.as_str())
+		.ok_or_else(|| Error::Config("expected `name`".into()))?;
+	Ok(name.to_string())
+}
+
+/// Get the current parachain id from the generated chain specification file.
+fn get_parachain_id(plain_parachain_spec: &Path) -> Result<u32> {
```

Comment:
```suggestion
fn get_parachain_id(chain_spec: &Path) -> Result<u32> {
```
This isnt specific to a plain chain spec - check the json for a raw chain spec and you will see the para_id still exists.